### PR TITLE
distinguish stdout and stderr

### PR DESCRIPTION
**Pull request by ndeloof**
**What I did**
Distinguish stdout and stderr when running exec in non-TTY mode

This code would better re-use https://github.com/docker/cli/blob/master/cli/command/container/exec.go#L74, need to make this a public func.

**Related issue**
closes https://github.com/docker/compose-cli/issues/1873

### CODE REVIEW

File: pkg/utils/writer.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ import (
 	"io"
 )
 
-// GetWriter creates a io.Writer that will actually split by line and format by LogConsumer
+// GetWriter creates a io.Stdout that will actually split by line and format by LogConsumer
```
Comment (mat007) [inline pkg/utils/writer.go:24]: This looks suspicious as well.

File: pkg/utils/writer.go [hunk 2]
```diff
@@ -34,7 +34,7 @@ type splitWriter struct {
 	consumer func(string)
 }
 
-// Write implements io.Writer. joins all input, splits on the separator and yields each chunk
+// Write implements io.Stdout. joins all input, splits on the separator and yields each chunk
```
Comment (mat007) [inline pkg/utils/writer.go:37]: This too.

**Review by mat007 — APPROVED** [review]
Looks good but for the changes in the comments (a bit too aggressive search and replace? 😁)

**Review by lorenrh — APPROVED** [review]

File: cli/server/proxy/streams/logs.go [hunk 3]
```diff
@@ -24,7 +24,7 @@ import (
 	containersv1 "github.com/docker/compose-cli/cli/server/protos/containers/v1"
 )
 
-// Log implements an io.Writer that proxies logs over a gRPC stream
+// Log implements an io.Stdout that proxies logs over a gRPC stream
```
[the remark this answers is not in the document]
Comment (ndeloof) [reply cli/server/proxy/streams/logs.go]: nope. Golang refactoring trying to be clever :P

**Pull request merged**